### PR TITLE
Add basics suite, refactor to share common infra

### PR DESCRIPTION
**Pull request by chefsalim**
This PR adds a new 'basics' suite - with a test that checks for unique gems.
It also refactors to pull out common functionality (KITCHEN env, YML files) into a shared folder.
This allows using a single set of switches to control the behavior of the various suites.

@chef/client-core @jkeiser @sersut

### CODE REVIEW

File: acceptance/shared/.kitchen.digitalocean.yml [hunk 1]
```diff
@@ -9,6 +9,12 @@ driver:
   transport:
     ssh_key: <%= ENV['DIGITALOCEAN_SSH_KEY_PATH'] %>
 
+provisioner:
+  name: chef_zero
+  product_name: chef
+  product_version: latest
+  channel: current
```
Comment (sersut) [inline acceptance/shared/.kitchen.digitalocean.yml]: Was it intentional here not to use the environment variables?

File: acceptance/basics/test/integration/chef-current-install/serverspec/chef_client_spec.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+
+require 'spec_helper'
+
+gem_path = "/opt/chef/embedded/bin/gem"
+white_list = %w{chef-config json rake}
```
Comment (sersut) [inline acceptance/basics/test/integration/chef-current-install/serverspec/chef_client_spec.rb]: Is it OK to have multiple versions of `chef-config` ?

**Comment by sersut** [comment]
This is cool. I would run an adhoc build through Jenkins to make sure we don't break test matrix.

`KITCHEN_CHEF_PRODUCT, KITCHEN_CHEF_VERSION, KITCHEN_CHEF_CHANNEL` are not set in Jenkins. So you might need to change [this](https://github.com/chef/chef/blob/master/ci/verify-chef.sh#L89) line in `ci/verify-chef.sh`.

File: acceptance/shared/.kitchen.digitalocean.yml (hunk 1 above)
Comment (chefsalim) [inline acceptance/shared/.kitchen.digitalocean.yml]: We're not currently supporting digital ocean so we haven't changed this file - it's still just a placeholder.

File: acceptance/basics/test/integration/chef-current-install/serverspec/chef_client_spec.rb (hunk 2 above)
Comment (chefsalim) [inline acceptance/basics/test/integration/chef-current-install/serverspec/chef_client_spec.rb]: Yeah, we apparently need multiple versions for compatibility.

File: acceptance/shared/.kitchen.digitalocean.yml (hunk 1 above)
Comment (sersut) [inline acceptance/shared/.kitchen.digitalocean.yml]: :+1:

File: acceptance/basics/test/integration/chef-current-install/serverspec/chef_client_spec.rb (hunk 2 above)
Comment (sersut) [inline acceptance/basics/test/integration/chef-current-install/serverspec/chef_client_spec.rb]: :+1:

File: acceptance/shared/libraries/kitchen.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
+class Kitchen < Chef::Resource
```
Comment (patrick-wright) [inline acceptance/shared/libraries/kitchen.rb]: I built something very similar.  I would like to move this to chef-acceptance so all projects benefit from it.

File: acceptance/shared/.kitchen.ec2.yml [hunk 4]
```diff
@@ -25,6 +25,12 @@ transport:
                file_if_exists("~/.ssh/#{ENV['AWS_SSH_KEY_ID'] || ENV['USER'] || ENV['USERNAME']}") ||
                file_if_exists("~/.ssh/id_rsa") %>
 
+provisioner:
+  name: chef_zero
+  product_name: <%= ENV["KITCHEN_CHEF_PRODUCT"] %>
```
Comment (patrick-wright) [inline acceptance/shared/.kitchen.ec2.yml]: ah memories :)  I feel like you guys are chef-acceptance gurus!

**Comment by patrick-wright** [comment]
After seeing a build run through jenkins :shipit:

**Pull request merged**